stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/Linkify.kt

```diff
@@ -0,0 +1,214 @@
+
+                // Fix capitalization if necessary
+                if (!url.regionMatches(0, prefixes[i], 0, prefixes[i].length, ignoreCase = false)) {
+                    url = prefixes[i] + url.substring(prefixes[i].length)
+                }
+                break
+            }
+        }
+        if (!hasPrefix && prefixes.isNotEmpty()) {
+            url = prefixes[0] + url
+        }
+        return url
+    }
+
+    private fun gatherLinks(
+        links: MutableList<LinkSpec>,
+        s: Spannable,
+        pattern: Pattern,
+        schemes: Array<String>,
+        matchFilter: Linkify.MatchFilter?,
+        transformFilter: Linkify.TransformFilter?,
+    ) {
+        val m = pattern.matcher(s)
+        while (m.find()) {
+            val start = m.start()
+            val end = m.end()
+            if (matchFilter == null || matchFilter.acceptMatch(s, start, end)) {
+                val url: String = makeUrl(m.group(0), schemes, m, transformFilter)
+                val spec = LinkSpec(url = url, start = start, end = end)
+                links.add(spec)
+            }
+        }
+    }
+
+    private fun pruneOverlaps(links: MutableList<LinkSpec>, text: Spannable) {
+        // Append spans added by Markwon to remove any overlap.
+        val urlSpans: Array<URLSpan> = text.getSpans(0, text.length, URLSpan::class.java)
+        urlSpans.forEach { span ->
+            val spec = LinkSpec(
+                markwonAddedSpan = span,
+                start = text.getSpanStart(span),
+                end = text.getSpanEnd(span)
+            )
+            links.add(spec)
+        }
+
+        links.sortWith(COMPARATOR)
+
+        var len = links.size
+        var i = 0
+        while (i < len - 1) {
+            val a: LinkSpec = links[i]
+            val b: LinkSpec = links[i + 1]
+            var remove = -1
+            if (a.start <= b.start && a.end > b.start) {
+                when {
+                    b.end <= a.end -> {
+                        remove = i + 1
+                    }
+                    a.end - a.start > b.end - b.start -> {
+                        remove = i + 1
+                    }
+                    a.end - a.start < b.end - b.start -> {
+                        remove = i
+                    }
+                }
+                if (remove != -1) {
+                    val span: URLSpan? = links[remove].markwonAddedSpan
+                    if (span != null) {
+                        text.removeSpan(span)
+                    }
+                    links.removeAt(remove)
+                    len--
+                    continue
+                }
+            }
+            i++
+        }
+    }
+
+    private data class LinkSpec(
+        val markwonAddedSpan: URLSpan? = null,
+        val url: String? = null,
+        val start: Int,
+        val end: Int,
+    )
+
+    private val DEFAULT_SPAN_FACTORY: (string: String?) -> URLSpan = { string: String? -> URLSpan(string) }
```

Comment:
No need to wrap the constructor inside a function
```suggestion
    private val DEFAULT_SPAN_FACTORY: (string: String?) -> URLSpan = ::URLSpan
```